lib/Conversion/TorchToLinalg/Reduction.cpp

```diff
@@ -322,6 +389,46 @@ class ConvertReductionOp : public ConversionPattern {
     auto resultType = getTypeConverter()
                           ->convertType(op->getResult(0).getType())
                           .cast<RankedTensorType>();
+
+    if (auto normOp = dyn_cast<AtenLinalgVectorNormOp>(op)) {
+      auto elemType = resultType.getElementType();
+
+      if (!elemType.isa<mlir::FloatType>())
+        return rewriter.notifyMatchFailure(
+            op, "only float types are valid for vector norm ops");
+
+      // Cast `ord` to float so that we can readily pass it math.powf.
+      auto ordValue = operands[1];
+      if (ordValue.getType().isa<mlir::IntegerType>())
+        ordValue = rewriter.create<arith::SIToFPOp>(loc, elemType, ordValue);
+
+      // Sum each element of the tensor after computing the exponentiation.
+      auto sumOp =
+          createNormSumReduction(loc, elemType, opInfo, ordValue, rewriter);
+      if (!sumOp)
+        return failure();
+
+      // TODO: Add support for L0 norm.
+      auto epsilon = 1e-5;
+      auto ordLiteral = 0.0;
+      if (matchPattern(ordValue, m_TorchConstantFloat(&ordLiteral)) &&
```

Comment:
`op.ord()` can be the integer `0`, which would not be caught by this guard.

Reply:
I'm not so sure.  Wouldn't the type conversion to `elemType` force `ordValue` to be a float here?